test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,552 @@ describe("BaseAxis", () => {
+              TestMethods.numAttr(firstAnnotationRect, "height"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows leftwards (orientation: left)", () => {
+            let axis = new Plottable.Axis(scale, "left");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "x") - TestMethods.numAttr(secondAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows rightwards (orientation: right)", () => {
+            let axis = new Plottable.Axis(scale, "right");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(secondAnnotationRect, "x") - TestMethods.numAttr(firstAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+        });
+
+      });
+
+      it("hides annotations if rectangles are outside the margin area", () => {
+        let mockScale = new Plottable.Scales.Linear;
+        let annotatedTicks = [50, 51, 150];
+        mockScale.domain(annotatedTicks);
+
+        let axis = new Plottable.Axis(mockScale, "bottom");
+        axis.annotatedTicks(annotatedTicks);
+        axis.annotationsEnabled(true);
+        axis.margin(25);
+
+        let svg = TestMethods.generateSVG(300, 300);
+        axis.renderTo(svg);
+
+        axis.content().selectAll(".annotation-rect").each(function() {
+          let annotationRect = d3.select(this);
+          let bbox = this.getBBox();
+          let insideMargin = bbox.x >= 0 && bbox.x + bbox.width <= axis.width() &&
+                             bbox.y >= axis.height() - axis.margin() && bbox.y + bbox.height <= axis.height();
+          let visibilityAttr = insideMargin ? "visible" : "hidden";
+          assert.strictEqual(annotationRect.attr("visibility"), visibilityAttr, "annotation rect inside margin area");
```

Comment:
Also, going to revive my earlier comment about semantic flow:
>This is a bit subtle, but changing the "semantic flow" of the test would better represent what is being tested. Currently, the test checks that the "insideness" and visibility of each annotation is the same. While that logic does check the functionality, the following logic would better represent the test title and what the test is looking for:
- Check if the rectangle is inside the margin.
  - If so, assert that it is visible
  - If not, assert that it is not visible

>Since this structure better mirrors the logical flow of the test's name, it's easier for developers to understand what it's doing and be confident that the test is checking the correct thing.